File: Level 1/Director of Photography 1/python/Director_of_Photography_1.py
def getArtisticPhotographCount(N: int, C: str, X: int, Y: int) -> int:
    count = 0
    for i in range(N - 2 * X):
        cur = C[i]
        if cur == 'P' or cur == 'B':
            opposite = 'B' if cur == 'P' else 'P'
            for j in range(i + X, min(i + Y + 1, N - X)):
                if C[j] == 'A':
                    for k in range(j + X, min(j + Y + 1, N)):
                        if C[k] == opposite:
                            count += 1
    return count

File: Level 1/Director of Photography 1/cpp/DirectorOfPhotography1.cpp
#include <string>

using namespace std;

int getArtisticPhotographCount(int N, string C, int X, int Y) {
  int count = 0;
  for (int i = 0; i < (N - 2 * X); i++) {
    char cur = C[i];
    if (cur == 'P' || cur == 'B') {
      char opposite = (cur == 'P') ? 'B' : 'P';
      for (int j = i + X; j <= (i + Y); j++) {
        if (j < (N - X) && C[j] == 'A') {
          for (int k = j + X; k <= (j + Y); k++) {
            if (k < N && C[k] == opposite) {
              count++;
            }
          }
        }
      }
    }
  }
  return count;
}